docs/source/etl/merge.rst

```diff
@@ -0,0 +1,94 @@
+into a multi-sample dataset. You can use Glow and Spark to perform this merge operation with a few
+lines of code. In these examples, we will read from VCF files, but the same logic works
+on DataFrames backed by other file formats.
+
+First, you need to read the VCF files into a single Spark DataFrame:
+
+.. code-block:: python
+
+  from pyspark.sql.functions import *
+
+  df = spark.read.format('vcf').load([path1, path2])
+  df.select('contigName', 'start', col('genotypes').sampleId)\
+    .orderBy('contigName','start').show()
+
+  # Alternatively, you can use the "union" DataFrame method
+  df1 = spark.read.format('vcf').load(path1)
+  df2 = spark.read.format('vcf').load(path2)
+  df = df1.union(df2)
+
+The resulting DataFrame contains all records from the VCFs you want to merge, but the genotypes from
+different samples at the same site have not been combined. You can use an aggregation to combine the
+genotype arrays:
+
+.. code-block:: python
+
+  from pyspark.sql.functions import *
+
+  merged_df = df.groupBy('contigName', 'start', 'end', 'referenceAllele', 'alternateAlleles')\
+    .agg(sort_array(flatten(collect_list('genotypes'))).alias('genotypes'))
+
+  merged_df.show()
+
+.. invisible-code-block:: python
+
+  from pyspark.sql import Row
+
+  row = merged_df.orderBy('contigName', 'start').select('contigName', 'start', 'genotypes.sampleId').head()
+  assert_rows_equal(row, Row(contigName='22', start=16050074, sampleId=['HG00096', 'HG00097']))
+
+The genotypes from different samples now appear in the same genotypes array.
+
+.. note::
+  
+  If the VCFs you are merging contain different sites, elements will be missing from the genotypes
+  array after aggregation. Glow automatically fills in missing genotypes when writing to
+  ``bigvcf``, so an exported VCF will still contain all samples.
+
+Aggregating INFO fields
+-----------------------
+
+To preserve INFO fields in a merge, you can use the aggregation functions in Spark. For instance, to
+emit an ``INFO_DP`` column that is the sum of the ``INFO_DP`` columns across all samples:
+
+.. code-block:: python
+
+  from pyspark.sql.functions import *
+
+  merged_df = df.groupBy('contigName', 'start', 'end', 'referenceAllele', 'alternateAlleles')\
+    .agg(sort_array(flatten(collect_list('genotypes'))).alias('genotypes'),
+         sum('INFO_DP').alias('INFO_DP'))
+
+  merged_df.select('contigName', 'start', 'INFO_DP').show()
+
+.. invisible-code-block:: python
+
+  row = merged_df.orderBy('contigName', 'start').select('contigName', 'start', 'genotypes.sampleId', 'INFO_DP').head()
+  assert_rows_equal(row, Row(contigName='22', start=16050074, sampleId=['HG00096', 'HG00097'],
+    INFO_DP=16024))
+  
+
+Joint genotyping
+----------------
+
+The merge logic in this document allows you to quickly aggregate genotype array data or single
+sample VCFs. For a more sophisticated aggregation that unifies alleles at overlapping sites and uses
```

Comment:
What kind of single-sample VCFs?

Reply:
I don't understand this question.

Reply:
I suppose I was asking if the single-sample VCFs had to come from the same source, which would imply they have the same header lines and schema.

Reply:
I think this is covered when we show how to read the VCFs. I don't want to presume too much about why people are merging VCFs because sometimes it really surprises me.